libs/javascript/@local/eslint-config/legacy-base-eslintrc-to-refactor.cjs

```diff
@@ -5,12 +5,10 @@ module.exports = {
   parser: "@typescript-eslint/parser",
   plugins: ["@typescript-eslint", "react-hooks", "jest", "unicorn"],
   extends: [
-    "plugin:@typescript-eslint/recommended-requiring-type-checking",
```

Comment:
We were linting `*.ts` and `*.tsx` files in CLI before. This meant that linting for all `*.js` and `*.cjs` files was broken, but we could only see this in the editor. Setting TS-specific rules to TS files only below fixes this problem.